notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
Currently the metrics are declared as global variables. This introduces problems as soon as someone tries to use this package independently somewhere. (It's often argued it's not going to happen – history shows it does anyway.) 

They are tied to the default (also global =/) registry of the Prometheus client library. Someone using the package but that has a custom Prometheus registry won't have access to those metrics.
If there's a data accessibility problem like in the alertmanagerSet, globals are often the easiest answer but generally a bad one.

Someone using two different instances of Notifier that have overlap in the labels they instantiate may also run into weird metric artifacts.

We are doing what you moved towards here in a couple of places but I'd generally like us to move away from it.
Using a separate `notifierMetrics` type is a somewhat easy way to have expressive addressability (`n.metrics.*`) of metrics and not polluting the main type.

Does this make sense?